### ISSUE TITLE
Multiple files upload with addToBody stores only one file

### ISSUE THREAD

**Issue by SkeLLLa** [fastify/fastify-multipart#64]
Hi. As far as can see from the code and examples if multiple files will be uploaded with singe file input, it will put them into the same field in body and `filename` and other fields inside it will correspond to one of the files.

Multiple files could be tracked by number of calls of `defaultConsumer` or if that function is called with different filename. So I think it's better to track it by call number which will represent an index in array. According to this field type in body could be also an array of items with `sharedSchemaId` type. And if that array has only one file, then, probably, it could replace array, so we'll get an file object as it's done right now.

**Comment by mcollina** [comment]
Definitely, would you like to send a PR?

**Comment by SkeLLLa** [comment]
Yes I can, probably in couple of days if someone will not make it faster.

But I have a question regarding filed type. If we have input with name `file` should the key `file` in body to be an array even if there's only one file in there. Or should we put an object instead? E.g.

```
body = {
  file: [{name, data, ...etc}]
}
```
or
```
body = {
  file: {name, data, ...etc}
}
```

Also I'm thinking about an idea of changing the default behaviour of this module. I think it would be better if this module will use `addToBody` behaviour as default one. It will be easier to use it and it will cover most of user usecases. Also it will make code look prettier, because all the stuff with event listeners, onFinish handlers will be done under the hood.

And if someone will need "streams API" I think it can be done in the same manner. But in field `file` for example we'll have and array of object where data will be a readable stream instead of buffer. And we'll add, for example, extra filed to request object that will be a promise, which can be awaited to ensure that all streams were read.

That's just some thoughts, that I think would be nice to have to release version 1.0.0.
WDYT, @mcollina.

**Comment by SkeLLLa** [comment]
I've made some progress regarding the PR, but I'm stuck in writing tests, because it seems that `form-data` doesn't support multiple values in same field, like browser does.

**Comment by nwoltman** [comment]
@SkeLLLa In what way does it not support multiple values? Appending to the same field should work.

```js
form.append('upload', file1)
form.append('upload', file2)
```

**Comment by SkeLLLa** [comment]
@nwoltman I've tried such approach, but in that way I receive only `file2` in upload field. Maybe the bug is somewhere in my changes, but when I use html form from browser all is working fine.

**Comment by Eomm** [comment]
> should the key file in body to be an array even if there's only one file in there. Or should we put an object instead?

I think that it should if we don't want to release a semver-major change.
I would like to keep the behaviour of one file = one field in the body and not an array (and multiple files in one field will be an array)

I don't know if this behaviour could be confusing for users, @mcollina WDYT?

**Comment by SkeLLLa** [comment]
Actually both options could be a bit confusing. If we have, for example one file = object, multiple files = array of objects, then it will need additional `if` statements on server side to check how many files were uploaded if there's input with multiple file upload enabled. On the other hand if we have an array of files in all cases then it will be a bit weird to have an array there if there is single file input.

**Comment by mcollina** [comment]
I’m ok in changing addToBody to set an array of files, it’s a pretty new feature and we won’t break much. I _might_ be ok in making addToBody the default. I’m not convinced in changes to the other API.

**Comment by nwoltman** [comment]
@SkeLLLa The solution to that would be to allow each route to configure how many files it expects for each specific file field. This is what [`multer`](https://github.com/expressjs/multer#singlefieldname) does.

**Comment by Eomm** [comment]
released in v1.0.0 👍
